### ISSUE TITLE
Validator: Additional rule parameters for built-in types

### ISSUE THREAD

**Issue by cdtweb** [nonamephp/php7-common#5]
# String parameters
* **allow_empty** : `bool` Allow an empty string value; Enabled by default
* **allow_null** : `bool` Allow a null value; Disabled by default
* **min_length** : `int` Minimum allowed `strlen` of value; Ignore by default
* **max_length** : `int` Maximum allowed `strlen` of value; Ignore by default
* **equals** : `string` Value must be equal to this value
* **in** : `array` Value must be in array
* **pattern**: `regex` Value must match regex pattern

# Integer parameters
* **unsigned** : `bool` Only allow unsigned integer values; Disabled by default
* **>**: `int` Value must be greater than this value
* **<** : `int` Value must be less than this value
* **>=** : `int` Value must be greater than or equal to this value
* **<=**: `int` Value must be less than or equal to this value
* **!=**: `int` Value must not be equal to this value
* **equals**: `int` Value must be equal to this value
* **in**: `array` Value must be in array

# Array parameters
* **count**: `int`Array must have exactly this many values
* **min_count**: `int` Array must have greater than or equal to this many values
* **max_count**: `int` Array must have less than or equal to this many values
* **allow_empty**: `bool` Allow an empty array; Enabled by default

## Basic Example: String Rule
```php
<?php
use Noname\Common\Validator;

$values = [
    'user_name' => 'john.doe'
];

$rules = [
    'user_name' => [
        'type' => 'str', 
        'allow_empty' => false, 
        'min_length' => 3, 
        'max_length' => 255
    ]
];

$validator = new Validator($values, $rules);
if(!$validator->validate()){
    $errors = $validator->getErrors();
    // handle errors
}
```

**Comment by cdtweb** [comment]
The following string modifiers need to be supported for 'string' type rules.

* **allow_empty** : bool
* **allow_null** : bool
* **min_length** : int
* **max_length** : int

**Comment by cdtweb** [comment]
As some of the rules may apply to different types (e.g. `>` will apply similarly to integers and numeric types) I need to think about how to avoid duplicating rule logic in each of the validate methods for types. 

Perhaps some sort of type inheritance may be in order, where `integer` could inherit from `numeric`, and `alpha` would inherit from `string`, etc.

**Update:** This will be achievable once issue #24 has been closed.

**Comment by cdtweb** [comment]
Changes released with version 0.3.2
